hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -399,6 +400,11 @@ object DataSourceWriteOptions {
     .defaultValue(1000)
     .withDocumentation("The number of partitions one batch when synchronous partitions to hive.")
 
+  val HIVE_SYNC_MODE: ConfigProperty[String] = ConfigProperty
+    .key("hoodie.datasource.hive_sync.mode")
+    .noDefaultValue()
```

Comment:
wondering if we can set jdbc as default.

Reply:
Also, can we introduce an ENUM for diff modes.

Reply:
what I meant, is lets make jdbc as default (since it spans spark datasource, spark-sql etc). From within spark-sql, may be we can make hms as default(if user does not explicitly set it)

Reply:
sounds good